Sources/SwiftFrameCore/Config/TextData.swift

```diff
@@ -0,0 +1,107 @@
+import AppKit
+import Foundation
+
+typealias AssociatedString = (string: String, data: TextData)
+
+struct TextData: Decodable, ConfigValidatable {
+
+    // MARK: - Properties
+
+    let titleIdentifier: String
+    let textAlignment: NSTextAlignment
+    /// Text group will be prioritized over this, if specified
+    let maxFontSizeOverride: CGFloat?
+    let fontOverride: FontSource?
+    let textColorOverrideString: String?
+    let groupIdentifier: String?
+    let topLeft: Point
+    let bottomRight: Point
+
+    internal private(set) var textColorOverride: NSColor?
+
+    var rect: NSRect {
+        let origin = CGPoint(x: topLeft.x, y: bottomRight.y)
+        return NSRect(origin: origin, size: CGSize(width: bottomRight.x - topLeft.x, height: topLeft.y - bottomRight.y))
+    }
+
+    // MARK: - Coding Keys
+
+    enum CodingKeys: String, CodingKey {
+        case titleIdentifier = "identifier"
+        case textColorOverrideString = "colorOverride"
+        case maxFontSizeOverride
+        case fontOverride = "customFontPath"
+        case topLeft
+        case bottomRight
+        case textAlignment = "alignment"
+        case groupIdentifier
+    }
+
+    // MARK: - Init
+
+    internal init(
+        titleIdentifier: String,
+        textAlignment: NSTextAlignment,
+        maxFontSizeOverride: CGFloat?,
+        fontOverride: FontSource?,
+        textColorOverrideString: String?,
+        groupIdentifier: String?,
+        topLeft: Point,
+        bottomRight: Point,
+        textColorOverride: NSColor? = nil) {
```

Comment:
Micro-nit: I'd put `{` on a new line here to better visually separate the signature from the body.